go/vt/vttablet/tabletmanager/restore.go

```diff
@@ -83,7 +83,24 @@ func (agent *ActionAgent) restoreDataLocked(ctx context.Context, logger logutil.
 	// Record local metadata values based on the original type.
 	localMetadata := agent.getLocalMetadataValues(originalType)
 	tablet := agent.Tablet()
-	dir := fmt.Sprintf("%v/%v", tablet.Keyspace, tablet.Shard)
+
+	keyspace := tablet.Keyspace
+	keyspaceInfo, err := agent.TopoServer.GetKeyspace(ctx, keyspace)
+	if err != nil {
+		return err
+	}
+	// For a SNAPSHOT keyspace, we have to look for backups of BaseKeyspace
+	// so we will pass the BaseKeyspace in RestoreParams instead of tablet.Keyspace
+	// Keyspace creation checks that BaseKeyspace is set for a SNAPSHOT keyspace,
+	// so we don't need to check again here
+	if keyspaceInfo.KeyspaceType == topodatapb.KeyspaceType_SNAPSHOT {
+		keyspace = keyspaceInfo.BaseKeyspace
+		// If we belong to a SNAPSHOT keyspace, let us disable_active_reparents
+		// and disable replication_reporter
+		*mysqlctl.DisableActiveReparents = false
```

Comment:
It seems unsafe to change these flags at runtime like this. With the possible exception of unit tests, flag variables should be thought of as immutable after `flag.Parse()`.

I think the right long-term solution would be to have an abstract "should I enable X?" that depends both on a flag value and on the keyspace type. However, I don't want to hold up the whole PITR feature on this; it's going to be non-trivial to safely combine flag values (known at process start-up) with topo data (must be read from a remote server).

For now, can we just go back to saying in docs that users are responsible for configuring the tablets in a snapshot keyspace with certain flag values like these? Sorry for going back and forth on this.

Reply:
ok :)